Assets/Scripts/Multiplayer/Photon/PhotonRPC.cs

```diff
@@ -0,0 +1,298 @@
+        {
+            Debug.Log($"Base command child count: {childCount}");
+            var parentCommand = FindParentCommand(parentGuid);
+            var command = new BaseCommand(parent: parentCommand);
+
+            AddPendingCommand(() => {}, commandGuid, parentGuid, command, childCount);
+        }
+
+        [Rpc(InvokeLocal = false)]
+        public static void RPC_BrushStrokeFull(NetworkRunner runner, NetworkedStroke strokeData, Guid commandGuid, Guid parentGuid = default, int childCount = 0)
+        {
+            var decode = NetworkedStroke.ToStroke(strokeData);
+
+            CreateBrushStroke(decode, commandGuid, parentGuid, childCount);
+        }
+
+        [Rpc(InvokeLocal = false)]
+        public static void RPC_BrushStrokeBegin(NetworkRunner runner, Guid id, NetworkedStroke strokeData, int finalLength)
+        {
+            var decode = NetworkedStroke.ToStroke(strokeData);
+
+            decode.m_Type = Stroke.Type.NotCreated;
+            decode.m_IntendedCanvas = App.Scene.MainCanvas;
+            
+            Array.Resize(ref decode.m_ControlPoints, finalLength);
+            Array.Resize(ref decode.m_ControlPointsToDrop, finalLength);
+
+            if(m_inProgressStrokes.ContainsKey(id))
+            {
+                Debug.LogError("Shouldn't be here!");
+                return;
+            }
+
+            m_inProgressStrokes[id] = decode;
+        }
+        
+        [Rpc(InvokeLocal = false)]
+        public static void RPC_BrushStrokeContinue(NetworkRunner runner, Guid id, int offset, NetworkedControlPoint[] controlPoints, bool[] dropPoints)
+        {
+            if(!m_inProgressStrokes.ContainsKey(id))
+            {
+                Debug.LogError("shouldn't be here!");
+                return;
+            }
+
+            var stroke = m_inProgressStrokes[id];
+            
+            for(int i = 0; i < controlPoints.Length; ++i)
+            {
+                stroke.m_ControlPoints[offset + i] = NetworkedControlPoint.ToControlPoint(controlPoints[i]);
+                stroke.m_ControlPointsToDrop[offset + i] = dropPoints[i];
+            }
+        }
+
+        [Rpc(InvokeLocal = false)]
+        public static void RPC_BrushStrokeComplete(NetworkRunner runner, Guid id, Guid commandGuid, Guid parentGuid = default, int childCount = 0)
+        {
+            if(!m_inProgressStrokes.ContainsKey(id))
+            {
+                Debug.LogError("shouldn't be here!");
+                return;
+            }
+
+            var stroke = m_inProgressStrokes[id];
+
+            CreateBrushStroke(stroke, commandGuid, parentGuid, childCount);
+
+            m_inProgressStrokes.Remove(id);
+        }
+
+        [Rpc(InvokeLocal = false)]
+        public static void RPC_DeleteStroke(NetworkRunner runner, int seed, Guid commandGuid, Guid parentGuid = default, int childCount = 0)
+        {
+            var foundStroke = SketchMemoryScript.m_Instance.GetMemoryList.Where(x => x.m_Seed == seed).First();
```

Comment:
"DuplicateStroke" calls a constructor on StrokeData that reuses m_Seed. This will need to be refactored.

Reply:
One last thought - is the way that seeds are generated safe from duplicates enough? I've never looked into the specifics of guids but the rng used for seed is kinda interesting. Collisions would give us a very elusive and hard to reproduce bug...